pyomo/solvers/plugins/solvers/cplex_direct.py

```diff
@@ -180,9 +180,15 @@ def _process_stream(arg):
                     if not _is_numeric(option):
                         raise
                     opt_cmd.set(float(option))
-            
+
+            self._rc = None
```

Comment:
This is only populated in cases of a raised `CplexSolverError` and will be `None` in cases of successful solves. Are we ok with that? I would have thought we should initialise it with the return code for a successful run instead (if it exists)

Reply:
Yeah good point. I presumed that the return code would always be `0` for a successful run and that any downstream code wouldn't find that particularly interesting. Will update anyway

Reply:
Actually, that's not quite correct. It **should** only be populated in case of a `CplexSolverError` because its actually misnamed; `self._error_code` is its true purpose. `Cplex.solve()` always returns `None` so there's no return code as such. Only an error code and a solution status code.

Reply:
OK, shall we rename to `self._error_code` then?